### PR TITLE
Code for Africa Our offices

### PR DESCRIPTION
**Pull request by koechkevin**
## Description

- Adds our offices block.
- Adds Location collection to cms for selection on our offices

Fixes # (issue)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Screenshots
image
Screenshot 2023-10-06 at 09 11 34

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: apps/codeforafrica/src/payload/collections/Locations.js
Comment (kelvinkipruto) [inline apps/codeforafrica/src/payload/collections/Locations.js:1]: Perhaps a better name would be `Offices`

File: apps/codeforafrica/payload.config.ts [hunk 1]
```diff
@@ -57,6 +58,7 @@ export default buildConfig({
     Donors,
     GuidingPrinciples,
     Impact,
+    Locations,
```
Comment (kelvinkipruto) [inline apps/codeforafrica/payload.config.ts:61]: Would a better option be to make it a global since it's a global setting?

Comment (kilemensi) [reply apps/codeforafrica/payload.config.ts:61]: We have more than one office so I think a `collection` `Offices` is alright.

Global would also work if we use an array of offices or something but collection is more straight forward for content editors as it works similar to pretty much every other content on the CMS.

File: apps/codeforafrica/src/lib/data/blockify/ourOffices.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    ...item,
+    position: {
+      lng: item?.location?.[0],
+      lat: item?.location?.[1],
+    },
+    center: {
+      lng: item?.location?.[0],
+      lat: item?.location?.[1],
+    },
+  }));
```
Comment (kilemensi) [inline apps/codeforafrica/src/lib/data/blockify/ourOffices.js:13]:
```suggestion
  const addresses = offices?.map((item) => {
    let location = null;
    if (item?.location?.length)
      location = { lng: item.location.[0], lat: item.location.[1] };
    }
    return {
      ...item,
      position: location,
      center: location,
    };
  });
```

File: apps/codeforafrica/src/components/OfficeAddresses/OfficeAddresses.js [hunk 3]
```diff
@@ -46,7 +50,7 @@ const OfficeAddresses = React.forwardRef(function OfficeAddresses(props, ref) {
           zIndex: -1,
         }}
       >
-        <GoogleMap {...map} {...activeAddress?.map} style={mapStyle} />
+        <GoogleMap {...map} {...activeAddress} style={mapStyle} />
```
Comment (kilemensi) [inline apps/codeforafrica/src/components/OfficeAddresses/OfficeAddresses.js:53]: Why passing `activeAddress`? Original code assumed `activeAddress` would have `map` & hence was passing `activeAddress.map`

File: apps/codeforafrica/src/components/AddressCard/AddressCard.js [hunk 4]
```diff
@@ -38,12 +47,37 @@ const AddressCard = React.forwardRef(function AddressCard(props, ref) {
...
+            variant="body3"
+          >
+            {addressLine2}
+          </RichTypography>
+          <RichTypography
+            sx={{ color: "inherit", "& p": { m: 0 } }}
+            variant="body3"
+            component="span"
+          >
+            {zipcode},
```
Comment (kilemensi) [inline apps/codeforafrica/src/components/AddressCard/AddressCard.js:67]: Only include `,` if there is `zipCode`. From collection definition, `zipCode` is not required and hence may be empty.

**Review by kilemensi — APPROVED** [review]

File: apps/codeforafrica/src/payload/collections/Offices.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
+const Offices = {
+  slug: "offices",
+  admin: {
+    group: "Settings",
+    defaultColumns: ["title"],
```
[the remark this answers is not in the document]
Comment (koechkevin) [reply apps/codeforafrica/src/payload/collections/Offices.js:5]: @kilemensi Can't  there be a situation where we have more than 1 office in a City?

**Pull request merged**